xls/modules/zstd/frame_header_test.cc

```diff
@@ -0,0 +1,379 @@
+  // https://datatracker.ietf.org/doc/html/rfc8878#name-window-descriptor
+  bool window_size_valid(uint64_t window_size) {
+    auto max_window_size =
+        (1 << TEST_WINDOW_LOG_MAX_LIBZSTD) +
+        (((1 << TEST_WINDOW_LOG_MAX_LIBZSTD) >> 3) * MAX_MANTISSA);
+
+    return window_size <= max_window_size;
+  }
+
+  void PrintZSTDFrameHeader(ZSTD_frameHeader* fh) {
+    std::cout << std::hex;
+    std::cout << "zstd_fh->frameContentSize: 0x" << fh->frameContentSize
+              << "\n";
+    std::cout << "zstd_fh->windowSize: 0x" << fh->windowSize << "\n";
+    std::cout << "zstd_fh->blockSizeMax: 0x" << fh->blockSizeMax << "\n";
+    std::cout << "zstd_fh->frameType: 0x" << fh->frameType << "\n";
+    std::cout << "zstd_fh->headerSize: 0x" << fh->headerSize << "\n";
+    std::cout << "zstd_fh->dictID: 0x" << fh->dictID << "\n";
+    std::cout << "zstd_fh->checksumFlag: 0x" << fh->checksumFlag << "\n";
+  }
+
+  // Form DSLX Value representing ZSTD Frame header based on data parsed with
+  // ZSTD library. Represents DSLX struct `FrameHeader`.
+  Value CreateExpectedFrameHeader(ZSTD_frameHeader* fh,
+                                  FrameHeaderStatus expected_status) {
+    if (expected_status == FrameHeaderStatus::CORRUPTED ||
+        expected_status == FrameHeaderStatus::UNSUPPORTED_WINDOW_SIZE) {
+      return Value::Tuple({
+          /*window_size:*/ Value(UBits(0, 64)),
+          /*frame_content_size:*/ Value(UBits(0, 64)),
+          /*dictionary_id:*/ Value(UBits(0, 32)),
+          /*content_checksum_flag: */ Value(UBits(0, 1)),
+      });
+    }
+    return Value::Tuple({
+        /*window_size:*/ Value(UBits(fh->windowSize, 64)),
+        /*frame_content_size:*/ Value(UBits(fh->frameContentSize, 64)),
+        /*dictionary_id:*/ Value(UBits(fh->dictID, 32)),
+        /*content_checksum_flag: */ Value(UBits(fh->checksumFlag, 1)),
+    });
+  }
+
+  // Create DSLX Value representing Buffer contents after parsing frame header
+  // in simulation. Represents DSLX struct `Buffer`.
+  Value CreateExpectedBuffer(Value dslx_simulation_input,
+                             absl::Span<const uint8_t> input_buffer,
+                             uint32_t consumed_bytes_count,
+                             FrameHeaderStatus expected_status) {
+    // Return original buffer contents
+    if (expected_status == FrameHeaderStatus::NO_ENOUGH_DATA) {
+      return dslx_simulation_input;
+    }
+    // Critical failure - return empty buffer
+    if (expected_status == FrameHeaderStatus::CORRUPTED ||
+        expected_status == FrameHeaderStatus::UNSUPPORTED_WINDOW_SIZE) {
+      return Value::Tuple({/*contents:*/ Value(UBits(0, dslx_buffer_size)),
+                           /*length:*/ Value(UBits(0, 32))});
+    }
+
+    // Frame Header parsing succeeded. Expect output buffer contents with
+    // removed first `consumed_bytes_count` bytes and extended to
+    // dslx_buffer_size if necessary
+    int bytes_to_extend =
```

Comment:
should we use `size_t` here? or make the conversion explicit?

Reply:
changed to `size_t`